src/lambertw.jl

```diff
@@ -0,0 +1,386 @@
+    cset(m,2,-1//3) # μ₂ literal in paper, but only in (4.22)
+    for i in 3:n-1  # coeffs are zero indexed
+        compa(i,m,a)
+        compm(i,m,a)
+    end
+    return m
+end
+
+const LAMWMU_FLOAT64 = lamwcoeff(Float64,500)
+
+function horner(x, p::AbstractArray,n)
+    n += 1
+    ex = p[n]
+    for i = n-1:-1:2
+        ex = :($(p[i]) + t * $ex)
+    end
+    ex = :( t * $ex)
+    Expr(:block, :(t = $x), ex)
+end
+
+function mkwser(name, n)
+    iex = horner(:x,LAMWMU_FLOAT64,n)
+    :(function ($name)(x) $iex  end)
+end
+
+eval(mkwser(:wser3, 3))
+eval(mkwser(:wser5, 5))
+eval(mkwser(:wser7, 7))
+eval(mkwser(:wser12, 12))
+eval(mkwser(:wser19, 19))
+eval(mkwser(:wser26, 26))
+eval(mkwser(:wser32, 32))
+eval(mkwser(:wser50, 50))
+eval(mkwser(:wser100, 100))
+eval(mkwser(:wser290, 290))
+
+# Converges to Float64 precision
+# We could get finer tuning by separating k=0,-1 branches.
+function wser(p,x)
+    x < 4e-11 && return wser3(p)
+    x < 1e-5 && return wser7(p)
+    x < 1e-3 && return wser12(p)
+    x < 1e-2 && return wser19(p)
+    x < 3e-2 && return wser26(p)
+    x < 5e-2 && return wser32(p)
+    x < 1e-1 && return wser50(p)
+    x < 1.9e-1 && return wser100(p)
+    x > 1/euler && @baddomain(x)  # radius of convergence
+    return wser290(p)  # good for x approx .32
+end
+
+# These may need tuning.
+function wser(p::Complex{T},z) where T<:Real
+    x = abs(z)
+    x < 4e-11 && return wser3(p)
+    x < 1e-5 && return wser7(p)
+    x < 1e-3 && return wser12(p)
+    x < 1e-2 && return wser19(p)
+    x < 3e-2 && return wser26(p)
+    x < 5e-2 && return wser32(p)
+    x < 1e-1 && return wser50(p)
+    x < 1.9e-1 && return wser100(p)
+    x > 1/euler && @baddomain(x)  # radius of convergence
+    return wser290(p)
+end
+
+@inline function _lambertw0(x) # 1 + W(-1/e + x)  , k = 0
+    ps = 2*euler*x;
+    p = sqrt(ps)
+    wser(p,x)
+end
+
+@inline function _lambertwm1(x) # 1 + W(-1/e + x)  , k = -1
+    ps = 2*euler*x;
+    p = -sqrt(ps)
+    wser(p,x)
+end
+
+"""
+    lambertwbp(z,k=0)
+
+Accurate value of `1 + W(-1/e + z)`, for `abs(z)` in `[0,1/e]` for `k` either `0` or `-1`.
+Accurate to Float64 precision for abs(z) < 0.32.
+If `k=-1` and `imag(z) < 0`, the value on the branch `k=1` is returned. `lambertwbp` is vectorized.
+
+```jldoctest
+julia> lambertw(-1/e + 1e-18, -1)
+-1.0
+
+julia> lambertwbp(1e-18, -1)
+-2.331643983409312e-9
+
+# Same result, but 1000 times slower
+julia> convert(Float64,(lambertw(-BigFloat(1)/e + BigFloat(10)^(-18),-1) + 1))
+-2.331643983409312e-9
+```
+
+!!! note
+    `lambertwbp` uses a series expansion about the branch point `z=-1/e` to avoid loss of precision.
+    The loss of precision in `lambertw` is analogous to the loss of precision
+    in computing the `sqrt(1-x)` for `x` close to `1`.
+"""
+function lambertwbp(x::Number,k::Integer)
+    k == 0 && return _lambertw0(x)
+    k == -1 && return _lambertwm1(x)
+    error("expansion about branch point only implemented for k = 0 and -1")
```

Comment:
An error regarding the arguments should be `throw(ArgumentError("..."))` instead of `error("...")`.

Reply:
Right. will do.